test-data/shift.yaml

```diff
@@ -1104,3 +1104,167 @@ post:
   - r2.type=number
   - r2.svalue=32
   - r2.uvalue=32
+
+---
+test-case: check that unsigned right shift by 0 is idempotent - svalue out of range
+
+pre:
+  - r0.type=number
+  #  This is 0x8000000000000000, which is out of range for a signed 64-bit number.
+  - r0.svalue=-9223372036854775808
+  - r0.uvalue=9223372036854775808
+  - r1.type=number
+  - r1.svalue=-9223372036854775808
+  - r1.uvalue=9223372036854775808
+
+code:
+  <start>: |
+    r0 >>= 0
+    if r1 == r0 goto <one>
+    r2 = 0
+    exit
+  <one>: |
+    r2 = 1
+    exit
+
+post:
+  - r0.type=number
+  - r0.svalue=-9223372036854775808
+  - r0.uvalue=9223372036854775808
+  - r1.type=number
+  - r1.svalue=-9223372036854775808
+  - r1.uvalue=9223372036854775808
+  - r0.svalue=r1.svalue
+  - r0.uvalue=r1.uvalue
+  - r2.svalue=1
+  - r2.type=number
+  - r2.uvalue=1
+
+messages:
+  - "1:2: Code is unreachable after 1:2"
+
+---
+test-case: check that unsigned right shift by 0 is idempotent - valid svalue
+
+pre:
+  - r0.type=number
+  #  This is 0x7fffffffffffffff, which is in range for a signed 64-bit number.
```

Comment:
the latter "which is in range..." phrase can be removed as it doesn't add any value in my opinion.